enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,496 @@
+> admission and conversion webhooks, aggregated API servers, finalizers) here in detail,
+> especially how they impact the OCP system architecture and operational aspects.
+> 
+> - For conversion/admission webhooks and aggregated apiservers: what are the SLIs (Service Level
+>   Indicators) an administrator or support can use to determine the health of the API extensions
+> 
+>   Examples (metrics, alerts, operator conditions)
+>   - authentication-operator condition `APIServerDegraded=False`
+>   - authentication-operator condition `APIServerAvailable=True`
+>   - openshift-authentication/oauth-apiserver deployment and pods health
+> 
+> - What impact do these API extensions have on existing SLIs (e.g. scalability, API throughput,
+>   API availability)
+> 
+>   Examples:
+>   - Adds 1s to every pod update in the system, slowing down pod scheduling by 5s on average.
+>   - Fails creation of ConfigMap in the system when the webhook is not available.
+>   - Adds a dependency on the SDN service network for all resources, risking API availability in case
+>     of SDN issues.
+>   - Expected use-cases require less than 1000 instances of the CRD, not impacting
+>     general API throughput.
+> 
+> - How is the impact on existing SLIs to be measured and when (e.g. every release by QE, or
+>   automatically in CI) and by whom (e.g. perf team; name the responsible person and let them review
+>   this enhancement)
+> 
+> - Describe the possible failure modes of the API extensions.
+> - Describe how a failure or behaviour of the extension will impact the overall cluster health
+>   (e.g. which kube-controller-manager functionality will stop working), especially regarding
+>   stability, availability, performance and security.
+> - Describe which OCP teams are likely to be called upon in case of escalation with one of the failure modes
+>   and add them as reviewers to this enhancement.
+
+_TBD: Do we need to describe anything here?_
+
+## Support Procedures
+
+### Conflicting CRDs
+
+If the Ingress Operator detects the presence of a conflicting version of the
+Gateway API CRDs, it updates the ingress clusteroperator to report a `Degraded`
+status condition with status `True` and a message explaining the situation:
+
+_TBD: Insert example output from `oc get clusteroperators/ingress -o yaml`._
+
+In this situation, the cluster-admin is expected to verify that workload would
+not be broken by handing life-cycle management of the CRDs over to the Ingress
+Operator:
+
+_TBD: Insert `oc` command to make the CRD ownership transition._
+
+Then the Ingress Operator takes ownership and updates the CRDs:
+
+_TBD: Insert example `oc get clusteroperators` and `oc get crds` commands._
+
+### Overriding the Ingress Operator
+
+_TBD: Should we describe how to turn off the Ingress Operator so that the
+cluster-admin can override the CRDs, or describe how Server-Side Apply enables
+the cluster-admin to take over the CRDs?_
```

Comment:
Sounds like you might need what we are doing for CAPI in this space?

Reply:
I'm hoping we don't need to implement an explicit override.  Even if the alternative is something a little complicated (involving SSA) or heavy-handed (such as using a CVO override and scaling down the whole operator), we just need something good enough for now and defer doing something better.